### PR TITLE
monika 1.7.2 (new formula)

### PR DESCRIPTION
**Pull request by raosan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/monika.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  desc "Synthetic monitoring made easy"
+  homepage "https://monika.hyperjump.tech"
+  url "https://registry.npmjs.org/@hyperjumptech/monika/-/monika-1.7.2.tgz"
+  sha256 "ec1e6a934ab1343c3dc5e576dcb73a78cdd6852f8353891889fec0ad4db18bf5"
+  license "MIT"
+
+  livecheck do
+    url :stable
+  end
+
```
Comment (chenrui333) [inline Formula/monika.rb:13]:
```suggestion
```

File: Formula/monika.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+
+    # Remove unexpected universal binaries
+    fsevents_dir = libexec/"lib/node_modules/@hyperjumptech/monika/node_modules/fsevents"
+    rm_rf fsevents_dir
```
Comment (chenrui333) [inline Formula/monika.rb:22]:
```suggestion
    # Replace universal binaries with native slices.
    deuniversalize_machos
```

File: Formula/monika.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  test do
+    output = shell_output("#{bin}/monika 2>&1", 1)
+    assert_match "Error: Configuration file not found. By default, Monika looks for monika.yml", output
```
Comment (chenrui333) [inline Formula/monika.rb:27]:
```suggestion
    (testpath/"config.yml").write <<~EOS
      notifications:
        - id: 5b3052ed-4d92-4f5d-a949-072b3ebb2497
          type: desktop
      probes:
        - id: 696a3f57-a674-44b5-8125-a62bd2709ac5
          name: 'test brew.sh'
          requests:
            - url: https://brew.sh
              body: {}
              timeout: 10000
    EOS

    monika_stdout = (testpath/"monika.log")
    pid = fork do
      $stdout.reopen(monika_stdout)
      exec bin/"monika", "-c", testpath/"config.yml"
    end
    sleep 2

    assert_match "Starting Monika. Probes: 1. Notifications: 1", monika_stdout.read
```

**Comment by raosan** [comment]
Thanks a lot for the code suggestion @chenrui333 , It really helped because i struggled with the syntax. I made additional changes too so to pass the pipelines tests. Now it's ready to be merged

**Review by SMillerDev — DISMISSED** [review]

**Comment by SMillerDev** [comment]
Thanks @raosan  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by SMillerDev** [comment]
Can you squash all your commits @raosan? CI doesn't like merge commits

**Review by SMillerDev — APPROVED** [review]

**Comment by chenrui333** [comment]
@raosan, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**